libraries/apollo-tooling/src/main/kotlin/com/apollographql/apollo3/tooling/IntrospectionCapability.kt

```diff
@@ -0,0 +1,103 @@
+package com.apollographql.apollo3.tooling
+
+import com.apollographql.apollo3.annotations.ApolloExperimental
+import com.apollographql.apollo3.tooling.IntrospectionCapability.DirectiveArgsIncludeDeprecated
+import com.apollographql.apollo3.tooling.IntrospectionCapability.DirectiveIsRepeatable
+import com.apollographql.apollo3.tooling.IntrospectionCapability.FieldArgsIncludeDeprecated
+import com.apollographql.apollo3.tooling.IntrospectionCapability.InputValueDeprecatedReason
+import com.apollographql.apollo3.tooling.IntrospectionCapability.InputValueIsDeprecated
+import com.apollographql.apollo3.tooling.IntrospectionCapability.SchemaDescription
+import com.apollographql.apollo3.tooling.IntrospectionCapability.TypeInputFieldsIncludeDeprecated
+import com.apollographql.apollo3.tooling.IntrospectionCapability.TypeIsOneOf
+import com.apollographql.apollo3.tooling.IntrospectionCapability.TypeSpecifiedByURL
+import com.apollographql.apollo3.tooling.graphql.PreIntrospectionQuery
+
+@ApolloExperimental
+enum class IntrospectionCapability {
+  /**
+   * `__Schema.description`, introduced in [October 2021](https://spec.graphql.org/October2021/)
+   */
+  SchemaDescription,
+
+  /**
+   * `__Type.specifiedByURL`, introduced in [October 2021](https://spec.graphql.org/October2021/)
+   */
+  TypeSpecifiedByURL,
+
+  /**
+   * `__Directive.isRepeatable`, introduced in [October 2021](https://spec.graphql.org/October2021/)
+   */
+  DirectiveIsRepeatable,
+
+  /**
+   * `__Type.inputFields`, `includeDeprecated` argument, introduced in [Draft](https://spec.graphql.org/draft/)
+   */
+  TypeInputFieldsIncludeDeprecated,
+
+  /**
+   * `__Directive.args`, `includeDeprecated` argument, introduced in [Draft](https://spec.graphql.org/draft/)
+   */
+  DirectiveArgsIncludeDeprecated,
+
+  /**
+   * `__Field.args`, `includeDeprecated` argument, introduced in [Draft](https://spec.graphql.org/draft/)
+   */
+  FieldArgsIncludeDeprecated,
+
+  /**
+   * `__InputValue.isDeprecated`, introduced in [Draft](https://spec.graphql.org/draft/)
+   */
+  InputValueIsDeprecated,
+
+  /**
+   * `__InputValue.deprecationReason`, introduced in [Draft](https://spec.graphql.org/draft/)
+   */
+  InputValueDeprecatedReason,
```

Comment:
I think these are all the same? Does it make sense to include one but not the others?

Reply:
More generally, the naming should be about "features" and not introspection names. Maybe `SpecifiedBy`, `OneOf`, `DeprecatedInputValues`, `RepeatableDirectives`, ...?

Reply:
In 830615e670e4c00db468d70facae9c689028cfa4 I've renamed the enum to` GraphQLFeature`, and made it less granular.